gulpfile.js

```diff
@@ -1,87 +1,101 @@
-		}));
-		this.push(null);
-	};
-	return src.pipe(gulp.dest(DIST));
+gulp.task("qext", () => {
+  var qext = {
+    name: "Variance Waterfall",
+    type: "visualization",
+    description: pkg.description + "\nVersion: " + VERSION,
+    version: VERSION,
+    icon: "waterfallchart",
+    preview: "qlik-variance-waterfall.png",
+    keywords: "qlik-sense, visualization",
+    author: pkg.author,
+    homepage: pkg.homepage,
+    license: pkg.license,
+    repository: pkg.repository,
+    dependencies: {
+      "qlik-sense": ">=5.5.x",
+    },
+  };
+  if (pkg.contributors) {
+    qext.contributors = pkg.contributors;
+  }
+  var src = require("stream").Readable({
+    objectMode: true,
+  });
+  src._read = function () {
+    this.push(
+      new gutil.File({
+        cwd: "",
+        base: "",
+        path: pkg.name + ".qext",
+        contents: Buffer.from(JSON.stringify(qext, null, 4)),
+      })
+    );
+    this.push(null);
+  };
+  return src.pipe(gulp.dest(DIST));
 });
 
-gulp.task('clean', function(){
-  return del([DIST], { force: true });
+gulp.task("clean", (done) => {
+  del([DIST], { force: true });
+  done();
 });
 
-gulp.task('zip-build', function(){
-  return gulp.src(DIST + '/**/*')
+gulp.task("zip-build", () => {
+  return gulp
+    .src(DIST + "/**/*")
     .pipe(zip(`${pkg.name}_${VERSION}.zip`))
     .pipe(gulp.dest(DIST));
 });
 
-gulp.task('add-assets', function(){
-  return gulp.src('./assets/**/*').pipe(gulp.dest(DIST));
+gulp.task("dev-build", () => {
+  return gulp
+    .src(DIST + "/**/*")
+    .pipe(gulp.dest(DIST));
 });
 
-gulp.task('webpack-build', done => {
+gulp.task("add-assets", () => {
+  return gulp.src("./assets/**/*").pipe(gulp.dest(DIST));
+});
+
+gulp.task("webpack-build", (done) => {
   webpack(webpackConfig, (error, statistics) => {
     const compilationErrors = statistics && statistics.compilation.errors;
-    const hasCompilationErrors = !statistics || (compilationErrors && compilationErrors.length > 0);
+    const hasCompilationErrors =
+      !statistics || (compilationErrors && compilationErrors.length > 0);
 
-    console.log(statistics && statistics.toString({ chunks: false, colors: true })); // eslint-disable-line no-console
+    console.log(
+      statistics && statistics.toString({ chunks: false, colors: true })
+    ); // eslint-disable-line no-console
 
     if (error || hasCompilationErrors) {
-      console.log('Build has errors or eslint errors, fail it'); // eslint-disable-line no-console
+      console.log("Build has errors or eslint errors, fail it"); // eslint-disable-line no-console
       process.exit(1);
     }
 
     done();
   });
 });
 
-gulp.task('build',
-  gulp.series('clean', 'webpack-build', 'qext', 'add-assets')
-);
 
-gulp.task('zip',
-  gulp.series('build', 'zip-build')
-);
+gulp.task("build", gulp.series("clean", "webpack-build", "qext", "add-assets"));
+
+const changeToSourceMap = (done) => {
+  webpackConfig.devtool = 'source-map';
+  done();
+};
```

Comment:
@haxxmaxx do you think this solution makes sense ? when build:dev is run it changes devtool to source-map in webpack config

Reply:
Well, maybe. The more common approach is that you have some type of script that you add a flag to, e.g. `--dev` in this case, and then you set the config accordingly. But this might be a valid work-around. I'll let @niekvanstaveren have a look at this, build scripts etc isn't my strongest area...